test/topology_random_failures/conftest.py

```diff
@@ -0,0 +1,28 @@
+#
+# Copyright (C) 2023-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+# This file configures pytest for all tests in this directory, and also
+# defines common test fixtures for all of them to use
+
+import pytest
+
+
+pytest_plugins = [
+    "test.topology.conftest",
+]
+
+
+def pytest_addoption(parser: pytest.Parser) -> None:
+    # Limit for number of tests in topology_experimental_raft/test_random_failures.  Can be set in pytest.ini
+    parser.addoption('--raft-failure-injections-tests-count', action='store', default=None, type=int,
```

Comment:
How is this different from `--repeat`?

Reply:
It's a number of random tests from the full matrix to run.

Reply:
If I understand correctly, when we run `pytest ... test_random_failures.py`, it will randomly select an `error_injection` and a `cluster_event` and run the `test_random_failures` function with those selected values.

If I increase this parameter to 10 in `pytest.ini`, it will select 10 random injections and 10 random events, and run `test_random_failures` for each pair of selection (?) so 10 times in total.

I will achieve the exact same effect by using `--repeat 10`, no?

So I don't understand the difference.

cc @kostja

Reply:
In general I'm afraid that you're adding a bunch of new parameters that nobody will know they exist (maybe I will know for a week and then forget), so nobody will use them.

We should not add parameters that are not strictly needed.

Reply:
These are not framework-wide parameters. They are just for this suite.

If `--raft-failure-injections-tests-count=10` then we will run 10 random pairs injections and events without repeats. `--repeat=10` will not run just unique pairs in general.

Reply:
My question still applies. Honestly tell me, who will use these parameters once the code is merged into `master`? If you expect core devs to use it -- they won't.

Maybe you are planning to create a new Jenkins job (parallel to our regular CI runs) that would run this suite with custom parameters in regular fashion? Maybe that would give reason for these parameters to exist.

I understand need for `seed` -- you need it to reproduce a failure.

But what about `index`? What do we need it for?

Reply:
@enaydanov I think this can be options in the beginning of the test file or test arguments with default values. I don't think anyone who understand what this test does will have a desire to configure it through pytest.ini - it can be well "configured" by patching the test itself.